libs/auth/src/java_card.ts

```diff
@@ -438,19 +455,33 @@ export class JavaCard implements Card {
    *
    * This wrapper should never throw errors.
    */
-  private async safeReadCardDetails(): Promise<Optional<CardDetails>> {
+  private async safeReadCardDetails(): Promise<CardDetails> {
     try {
       return await this.readCardDetails();
-    } catch {
-      return undefined;
+    } catch (error) {
+      const errorMessage = extractErrorMessage(error);
+      const reason = (() => {
+        if (errorMessage.includes('certificate has expired')) {
```

Comment:
Openssl shell commands don't seem to return reliably documented error codes, nor do we have code in place to propagate such error codes, so we inspect error messages

Reply:
This feels fragile. If we want to keep this instead of using the OpenSSL API directly, could we at least do a version check and assert that it's an acceptable version?

Reply:
I like the version check idea! I'll also add that we'll get another layer of safety once Adam and I complete https://github.com/votingworks/vxsuite/issues/4567, which will be making it into the release. To run OpenSSL in FIPS compliant mode, we're going to be building OpenSSL from source. That'll be part of Trusted Build, so in production, we'll be pinned to an exact, reproducible build of OpenSSL. And I'm planning to keep dev and prod consistent here, so that build from source will be part of dev env setup, too.

An automated test for this case would also help, but as noted in the code comment you later reference, it's hard to create test certs with start dates in the future. There are ways to do it to be clear, but they require either messing around with the system time using `sudo`, which doesn't lend itself nicely to a script that can be used by others, or switching the cert creation logic from `openssl x509` to `openssl ca`, which I don't want to do this late in the game. But would like to update post cert freeze!